Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -351,4 +351,40 @@ public struct DiagnosticsFormatter {
       suffixTexts: [:]
     )
   }
+
+  /// Produce a string containing "footnotes" for each of the diagnostic
+  /// category provided that has associated documentation. Each category
+  /// is printed in Markdown link format, e.g.,
+  ///
+  /// ```
+  /// [#categoryName]: <categoryDocumentationURL>
+  /// ```
+  ///
+  /// This function also deduplicates entries and alphabetizes the results.
+  ///
+  /// - Parameters:
+  ///   - categories: the categories to print
+  ///   - leadingText: text that is prefixed to the list of categories when
+  ///     there is at least one category to print.
+  public func categoryFootnotes(
+    _ categories: [DiagnosticCategory],
```

Comment:
Could we make this `some Collection<DiagnosticCategory>`, say if somebody collected the categories in a `Set` (which doesn’t seem too unreasonable to me).

Reply:
Great idea! https://github.com/swiftlang/swift-syntax/pull/3008